egg/disk_management/disk.py

```diff
@@ -0,0 +1,34 @@
+import parted
+from parted import Device
+
+from egg.disk_management.partition import Partition
+
+
+class Disk(object):
+    def __init__(self, device: Device = None, device_path: str = None) -> None:
+        if device is not None:
+            self.device: parted.Device = device
+        elif device_path:
+            self.device: parted.Device = parted.getDevice(device_path)
+        else:
+            raise parted.DiskException("no device specified")
+        self.disk: parted.Disk = parted.newDisk(self.device)
+        self.partitions: list = list(map(lambda x: Partition(x), self.disk.partitions))
+
+    def __del__(self) -> None:
+        self.device.removeFromCache()
+
+    def get_model(self) -> str:
+        return self.device.model.__str__()
+
+    def get_type(self) -> str:
```

Comment:
Just a suggestion:

Python doesn't have a notion of private fields (everything is public), therefore getters are not frequently used. If you still need to have some code executed when a field is accessed, you can use the `@property` decorator, like:

```python
@property
def type(self) -> str:
    return self.disk.type
``` 

which can then be used like `my_object.type`